packages/common/container-definitions/src/runtime.ts

```diff
@@ -102,6 +102,9 @@ export interface IRuntime extends IDisposable {
      * @param snapshot - snapshot created at attach time
      */
     notifyAttaching(snapshot: ISnapshotTreeWithBlobContents): void;
+
+    // TODO
+    close(error?: Error): void;
```

Comment:
I'm not sure what's the purpose of this method. If I got it right, here are the interesting flows:
1. Container runtime may hit an error - it should request container to close, but not dispose.
2. Container closes - it does a bunch of stuff that is fully supported today by runtime (raises "readonly", "disconnect" events, etc.).
3. Container disposes - same as above (calls dispose() on container runtime, does not raise "readonly" event).

It feels like none of those workflows are asking for extra method on container runtime.